conduit-extra/Data/Conduit/Process.hs

```diff
@@ -60,6 +63,47 @@ sourceProcessWithConsumer cp consumer = do
 sourceCmdWithConsumer :: MonadIO m => String -> Consumer ByteString m a -> m (ExitCode, a)
 sourceCmdWithConsumer cmd = sourceProcessWithConsumer (shell cmd)
 
+
+-- | Given a @CreateProcess@, run the process
+-- and feed the provided @Producer@
+-- to the stdin @Sink@ of the process.
+-- Use the process outputs (stdout, stderr) as @Source@s
+-- and feed it to the provided @Consumer@s.
+-- Once the process has completed,
+-- return a tuple of the @ExitCode@ from the process
+-- and the results collected from the @Consumer@s.
+--
+-- IO is required because the streams are run concurrently
+-- using the <https://hackage.haskell.org/package/async async> package
+sourceProcessWithStreams :: CreateProcess
+                         -> Producer IO ByteString   -- ^stdin
+                         -> Consumer ByteString IO a -- ^stdout
+                         -> Consumer ByteString IO b -- ^stderr
+                         -> IO (ExitCode, a, b)
+sourceProcessWithStreams cp producerStdin consumerStdout consumerStderr = do
+    (  (sinkStdin, closeStdin)
+     , (sourceStdout, closeStdout)
+     , (sourceStderr, closeStderr)
+     , cph) <- streamingProcess cp
+    (_, resStdout, resStderr) <-
+      runConcurrently $ (,,)
```

Comment:
Question: what behavior do we want if the provided `Source` or `Sink`s throw an exception? It may be appropriate to have an `onException` here that closes the streams and kills the child process. Thoughts?